### PR TITLE
Fix 2798

### LINKED ISSUE #2798: nikola new_post should guess formats from extensions
* Running `nikola new_post -i /path/to/foo.ipynb` will create something like `posts/foo-title.rst`, which will result in completely broken output (turns out reST isn’t really good at rendering JSON not in a code block)
* Running `nikola new_post posts/bar.ipynb` will create reST-style metadata in that file, instead of a JSON file for ipynb.

(Replace *reST, ipynb* with *default compiler, any other compiler*)

### PR DESCRIPTION
**Pull request by ralsina**
Guess content format from path or imported file on `nikola new_post`

Fixes #2798

### CODE REVIEW

File: nikola/plugins/command/new_post.py [hunk 1]
```diff
@@ -265,16 +265,33 @@ def _execute(self, options, args):
...
+                    content_format = compiler
+
+        elif not content_format:  # Issue #400
             content_format = get_default_compiler(
                 is_post,
                 self.site.config['COMPILERS'],
                 self.site.config['post_pages'])
 
-        if content_format not in compiler_names:
+        elif content_format not in compiler_names:
```
Comment (Kwpolska) [inline nikola/plugins/command/new_post.py]: Perhaps that `if` was there for a reason? (I can’t see it though.)

File: nikola/plugins/command/new_post.py [hunk 2]
```diff
@@ -265,16 +265,33 @@ def _execute(self, options, args):
...
-        if not content_format:  # Issue #400
+        if not content_format and path:
+            # content_format not specified. If path was given, use
+            # it to guess (Issue #2798)
+            extension = os.path.splitext(path)[-1]
+            for compiler, extensions in self.site.config['COMPILERS'].items():
+                if extension in extensions:
+                    content_format = compiler
+
+        elif not content_format and import_file:
```
Comment (Kwpolska) [inline nikola/plugins/command/new_post.py]: Perhaps we could merge that and the first if into one?

File: nikola/plugins/command/new_post.py (hunk 1 above)
Comment (ralsina) [inline nikola/plugins/command/new_post.py]: I checked, this seems to work just as well :)

File: nikola/plugins/command/new_post.py (hunk 2 above)
Comment (ralsina) [inline nikola/plugins/command/new_post.py]: Perhaps, but it saves 3 lines of code at the cost of a non-stupid condition. I like stupid conditions.

**Comment by Kwpolska** [comment]
Thanks for fixing this!

**Pull request merged**